curp/src/server/bg_tasks.rs

```diff
@@ -81,40 +70,11 @@ pub(super) async fn run_bg_tasks<
     let bg_get_sync_cmds_handle =
         tokio::spawn(bg_get_sync_cmds(Arc::clone(&state), sync_chan, ae_trigger));
 
-    // spawn cmd execute worker
-    let cmd_executor = Arc::new(cmd_executor);
-    let bg_exe_worker_handles: Vec<JoinHandle<_>> = iter::repeat((
-        cmd_exe_rx,
-        Arc::clone(&spec),
-        Arc::clone(&cmd_board),
-        Arc::clone(&cmd_executor),
-    ))
-    .take(N_EXECUTE_WORKERS)
-    .map(|(rx, spec_c, cmd_board_c, ce)| tokio::spawn(execute_worker(rx, cmd_board_c, spec_c, ce)))
-    .collect();
-    let bg_as_worker_handles: Vec<JoinHandle<_>> = iter::repeat((
-        cmd_as_rx,
-        Arc::clone(&spec),
-        Arc::clone(&cmd_board),
-        cmd_executor,
-    ))
-    .take(N_AFTER_SYNC_WORKERS)
-    .map(|(rx, spec_c, cmd_board_c, ce)| {
-        tokio::spawn(after_sync_worker(rx, cmd_board_c, spec_c, ce))
-    })
-    .collect();
-
     shutdown.recv().await;
     bg_tick_handle.abort();
     bg_ae_handle.abort();
     bg_apply_handle.abort();
     bg_get_sync_cmds_handle.abort();
-    for handle in bg_exe_worker_handles {
```

Comment:
How can we abort these worker tasks when a `Protocol` struct is dropped?

Reply:
Fixed.